Sources/Basic/OutputByteStream.swift

```diff
@@ -521,3 +523,73 @@ public final class BufferedOutputByteStream: OutputByteStream {
         contents += bytes
     }
 }
+
+/// Represents a stream which is backed to a file. Not for instantiating.
+public class FileOutputByteStream: OutputByteStream {
+
+    /// Closes the file flushing any buffered data.
+    public func close() throws {
+        fatalError("close() should be implemented by a subclass")
+    }
+}
+
+/// Implements file output stream for local file system.
+public final class LocalFileOutputByteStream: FileOutputByteStream {
+
+    /// The pointer to the file.
+    let fp: UnsafeMutablePointer<FILE>
+
+    /// True if there were any IO error during writing.
+    private var error: Bool = false
+
+    /// Instantiate using the file pointer.
+    init(fp: UnsafeMutablePointer<FILE>) throws {
```

Comment:
Yeah, I didn't think we should expose something POSIX so I left it internal, But i'll change the name